plugins/modules/system/java_keystore.py

```diff
@@ -423,26 +469,50 @@ def create(self):
         else:
             self.openssl_create_pkcs12_bundle(keystore_p12_path)
 
+        if self.keystore_type == 'pkcs12':
+            # Preserve properties of the destination file, if any.
+            self.module.atomic_move(keystore_p12_path, self.keystore_path)
+            self.update_permissions()
+            self.result['changed'] = True
+            return self.result
+
         import_keystore_cmd = [self.keytool_bin, "-importkeystore",
                                "-destkeystore", self.keystore_path,
                                "-srckeystore", keystore_p12_path,
                                "-srcstoretype", "pkcs12",
                                "-alias", self.name,
                                "-noprompt"]
 
-        (rc, import_keystore_out, dummy) = self.module.run_command(
+        if self.keystore_type == 'jks':
+            keytool_help = self.module.run_command([self.keytool_bin, '-importkeystore', '-help'])
+            if '-deststoretype' in keytool_help[1] + keytool_help[2]:
+                import_keystore_cmd.insert(4, "-deststoretype")
+                import_keystore_cmd.insert(5, self.keystore_type)
+
+        keystore_backup = None
+        if self.exists():
+            keystore_backup = self.keystore_path + '.tmpbak'
```

Comment:
Seems like there is a module [method](https://github.com/ansible/ansible/blob/58f26388be7fc20aee8a3f43863c7832eea21fb6/lib/ansible/module_utils/basic.py#L1596-L1610) for this which might help avoid name conflicts and add predictability on repeat failures when creating backups, but this works as well.